### PR TITLE
Add test to verify source integrity

### PR DESCRIPTION
**Pull request by flodolo**

### CODE REVIEW

File: tests/integrity/source_integrity.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    foreach ($website_files as $filename => $locales) {
+        // Check if the file is known
+        if (! in_array($filename, $supported_files)) {
+            $errors[] = "Unknown file in array \$langfiles_subsets: {$filename}";
+        }
+        // Check if supported locales are known
+        $unknown_locales = array_diff($locales, $supported_locales);
+        if (! empty($unknown_locales)) {
+            $errors[] = "File $filename for website {$website_name} in \$langfiles_subsets supports unknown locales: " . implode(', ', $unknown_locales) . '.';
```
Comment (TheoChevalier) [inline tests/integrity/source_integrity.php]: {$filename}

File: tests/integrity/source_integrity.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    $website_files = $website_data[4];
+    if (empty($website_files)) {
+        $errors[] = "Website with ID {$website_id} ({$website_name}) doesn't include any file.";
+    } else {
+        $supported_files = array_merge($supported_files, $website_files);
+        // Check flags if available
+        if (! empty($website_data[7])) {
+            foreach ($website_data[7] as $filename => $flags) {
+                if (! in_array($filename, $website_files)) {
+                    $errors[] = "Website with ID {$website_id} ({$website_name}) has a flag for *$filename* but the file is not part of the website.";
```
Comment (TheoChevalier) [inline tests/integrity/source_integrity.php]: I guess you want {$filename} here as well?

File: tests/integrity/source_integrity.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+// Check websites
+$supported_files = [];
+$supported_locales = array_merge($mozilla, ['en-US']);
+$supported_websites = [];
+foreach ($sites as $website_id => $website_data) {
+    $website_name = $website_data[0];
+    $supported_websites[] = $website_name;
+
+    // Check supported Locales
```
Comment (TheoChevalier) [inline tests/integrity/source_integrity.php]: nit: locales

File: tests/integrity/source_integrity.php (hunk 2 above)
Comment (flodolo) [inline tests/integrity/source_integrity.php]: Yep

**Comment by TheoChevalier** [comment]
Don’t see anything else, works fine :tada:

**Comment by flodolo** [comment]
Thanks.

**Pull request merged**